### PR TITLE
[libleidenalg] fix broken download until upstream is ready

### PR DESCRIPTION
**Pull request by theAeon**
Fixes https://github.com/microsoft/vcpkg/pull/28734#issuecomment-1397795308

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
IMO it would be better to not issue the 0.9.0 tag again, even if it was lost. Would 0.9.0.1 be acceptable?

**Comment by theAeon** [comment]
I suppose, but the only difference between this and the lost 0.9.0 is a typo in a cmake file. Seems like a decent candidate for a port bump (especially as the REAL real 0.9.0 is from when the python library was the only one, so it's all fake.)

**Comment by theAeon** [comment]
@vtraag do you know how you're planning to tag the first c++ release proper?

File: ports/libleidenalg/portfile.cmake [hunk 1]
```diff
@@ -3,9 +3,9 @@ vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO theAeon/libleidenalg
-    REF "v${VERSION}"
-    SHA512 8ce68de2d16462dbeb8d6d0bf7740e20d7977da6da4dff953e60f9dfc5964149be0fb85026a49d0b41589af4ef3454c64253c6a6f6fdaa84e460b0756896fb42
-    HEAD_REF master
+    REF "${VERSION}"
```
Comment (BillyONeal) [inline ports/libleidenalg/portfile.cmake:6]: Since upstream has broken this before can this be changed to a commit sha instead?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/libleidenalg/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "libleidenalg",
   "version": "0.9.0",
+  "port-version": 2,
```
Comment (BillyONeal) [inline ports/libleidenalg/vcpkg.json:4]:
```suggestion
  "port-version": 1,
```

File: ports/libleidenalg/portfile.cmake (hunk 1 above)
Comment (theAeon) [reply ports/libleidenalg/portfile.cmake:6]: upstream never broke this b/c upstream was never ready for the vcpkg port-I broke it when I deleted the initial fork to re-fork from the upstream c++. Won't be an issue in the future.

Comment (BillyONeal) [reply ports/libleidenalg/portfile.cmake:6]: Sorry, by "upstream" I mean "upstream of vcpkg" which in this case is theAeon/libleidenalg :)

Comment (theAeon) [reply ports/libleidenalg/portfile.cmake:6]: ah i see-i don't plan on touching this until vtraag's repo is ready to be used as the source, so I don't see this being a problem.

**Review by BillyONeal — DISMISSED** [review]
I just applied the change to use a ref.

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**